core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -1145,17 +1150,22 @@ object GroupMetadataManager {
    * @param buffer input byte-buffer
    * @return an OffsetKey or GroupMetadataKey object from the message
    */
-  def readMessageKey(buffer: ByteBuffer): BaseKey = {
+  def readMessageKey(buffer: ByteBuffer): Option[BaseKey] = {
```

Comment:
Have you considered introducing an `UnknownKey` which extends `BaseKey`? The `UnknownKey` could carry on the version and even the bytes. It seems that the change would be less disruptive like this.